app/src/main/java/helium314/keyboard/keyboard/KeyboardActionListenerImpl.kt

```diff
@@ -143,16 +152,15 @@ class KeyboardActionListenerImpl(private val latinIME: LatinIME, private val inp
         val current = RichInputMethodManager.getInstance().currentSubtype.rawSubtype
         var wantedIndex = subtypes.indexOf(current) + if (steps > 0) 1 else -1
         wantedIndex %= subtypes.size
-        if (wantedIndex < 0)
-            wantedIndex += subtypes.size
+        if (wantedIndex < 0) wantedIndex += subtypes.size
         val newSubtype = subtypes[wantedIndex]
 
         // do not switch if we would switch to the initial subtype after cycling all other subtypes
-        if (initialSubtype == null)
-            initialSubtype = current
+        if (initialSubtype == null) initialSubtype = current
         if (initialSubtype == newSubtype) {
-            if ((subtypeSwitchCount > 0 && steps > 0) || ((subtypeSwitchCount < 0 && steps < 0)))
+            if (subtypeSwitchCount > 0 && steps > 0 || subtypeSwitchCount < 0 && steps < 0) {
```

Comment:
I know the logical and / or order is defined, but I still much prefer having parentheses for clarification.